packages/shared/lib/integrations/scripts/webhook/github-app-webhook-routing.ts

```
@@ -0,0 +1,116 @@
+import type { InternalNango as Nango } from './webhook.manager.js';
+import get from 'lodash-es/get.js';
+import type { Config as ProviderConfig } from '../../../models/Provider.js';
+import type { AuthCredentials } from '../../../models/Auth.js';
+import type { Connection, ConnectionConfig } from '../../../models/Connection.js';
+import connectionService from '../../../services/connection.service.js';
+import environmentService from '../../../services/environment.service.js';
+import { updateSuccess as updateSuccessActivityLog, createActivityLogMessageAndEnd } from '../../../services/activity/activity.service.js';
+import configService from '../../../services/config.service.js';
+import { connectionCreated as connectionCreatedHook } from '../../../hooks/hooks.js';
+import crypto from 'crypto';
+
+function validate(integration: ProviderConfig, headerSignature: string, body: any): boolean {
+    const hash = `${integration.oauth_client_id}${integration.oauth_client_secret}${integration.app_link}`;
+    const secret = crypto.createHash('sha256').update(hash).digest('hex');
+
+    const signature = crypto.createHmac('sha256', secret).update(JSON.stringify(body)).digest('hex');
+
+    const trusted = Buffer.from(`sha256=${signature}`, 'ascii');
+    const untrusted = Buffer.from(headerSignature, 'ascii');
+
+    return crypto.timingSafeEqual(trusted, untrusted);
+}
+
+export default async function route(_nango: Nango, integration: ProviderConfig, headers: Record<string, any>, body: any) {
+    const signature = headers['x-hub-signature-256'];
+
+    if (signature) {
+        console.log('Signature found, verifying...');
+        const valid = validate(integration, signature, body);
+
+        if (!valid) {
+            console.log('Github App webhook signature invalid');
+            return;
```

should it `reject` the promise instead of returning void?

---

I don't think so here. We're not expecting anything in particular from the call so it can just return `void`